src/Microsoft.AspNet.Mvc.Core/ActionResults/ViewExecutor.cs

```diff
@@ -64,23 +63,10 @@ public static class ViewExecutor
 
             response.ContentType = contentTypeHeader.ToString();
 
-            var wrappedStream = new StreamWrapper(response.Body);
-
-            using (var writer = new StreamWriter(wrappedStream, encoding, BufferSize, leaveOpen: true))
+            using (var writer = new HttpResponseStreamWriter(response.Body, encoding))
             {
-                try
-                {
-                    var viewContext = new ViewContext(actionContext, view, viewData, tempData, writer);
-                    await view.RenderAsync(viewContext);
-                }
-                catch
-                {
-                    // Need to prevent writes/flushes on dispose because the StreamWriter will flush even if
```

Comment:
This is a big concern, and something we should try to address in our writer.

If we flush on dispose then the headers + empty body get sent to the client. Then it means that a diagnostic middleware or exception filter can't write anything.